databricks/databricks-sdk-java
# Fix databricks CLI authentication on Windows

## Changes
There are two problems with respect to the Java SDK's auth pathways:
1. The logic to find the `.databrickscfg` file only works on *nix, as it hard-codes the path using forward-slashes. I've rewritten this to use `Paths.get`.
2. The databricks binary on Windows is called `databricks.exe`, not `databricks`, so we need to search for the binary with OS-specific filename.

To make this change unit-testable, I introduce an `Environment` class to abstract over environment variables, the search path for the databricks CLI executable, and the operating system name. This is needed because there are both differences in the environment variable name (`PATH` in Linux/Mac vs `Path` in Windows) and in the path separator (`:` in Linux/Mac vs `;` in Windows). The only system-independent way to get the path seems to be `System.getenv("PATH")` which works even on Windows. Note that `System.getenv().get("PATH")` doesn't work on Windows. Additionally, determining the OS depends on `System.getProperty`, so this can be initialized directly in DatabricksConfig and mocked in unit tests.

## Tests
Manually tested on a Windows machine:
1. Installed the Databricks CLI and ran `.\datarbricks.exe auth login --host <HOST>`.
2. Added the Databricks CLI to the PATH system environment variable.
3. Ran the CLIWorkspaceAuth example and was able to list clusters.

As a side note: when we make auth permutations test harnesses, we should have a Windows variant to catch these errors.
